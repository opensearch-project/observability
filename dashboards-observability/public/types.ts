/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { DashboardStart } from '../../../src/plugins/dashboard/public';
import { NavigationPublicPluginStart } from '../../../src/plugins/navigation/public';

export interface AppPluginStartDependencies {
  navigation: NavigationPublicPluginStart;
  dashboard: DashboardStart;
}

export interface ObservabilitySetup {}

export interface ObservabilityStart {}
