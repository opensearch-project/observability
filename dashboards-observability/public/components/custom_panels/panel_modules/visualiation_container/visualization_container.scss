/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

 .mouseGrabber {
  & :hover {
    cursor: -webkit-grab;
    cursor: grab;
  }
  & :active {
    cursor: -webkit-grabbing;
    cursor: grabbing;
  }
}

.visualization-div {
  width: 100%;
  height: 90%;
  overflow: scroll;
  text-align: center;
}

%center-div {
  top: 50%;
  left: 50%;
  -ms-transform: translate(-50%, -50%);
  -moz-transform: translate(-50%, -50%);
  -webkit-transform: translate(-50%, -50%);
  -o-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
}

.visualization-loading-chart {
  margin: 0;
  position: absolute;
  @extend %center-div;
}

.visualization-error-div {
  overflow: scroll;
  position: relative;
  @extend %center-div;
}

.panel-full-width {
  width: 100%;
  height: 100%;
}
