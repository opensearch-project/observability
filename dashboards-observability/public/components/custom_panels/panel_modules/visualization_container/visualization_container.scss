/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

.mouseGrabber {
  & :hover {
    cursor: -webkit-grab;
    cursor: grab;
  }
  & :active {
    cursor: -webkit-grabbing;
    cursor: grabbing;
  }
}

.visualization-action-button {
  & :hover {
    cursor: -webkit-pointer;
    cursor: pointer;
  }
}

.visualization-div {
  width: 100%;
  height: 90%;
  overflow: auto;
  text-align: center;
}

%center-div {
  top: 50%;
  left: 50%;
  -ms-transform: translate(-50%, -50%);
  -moz-transform: translate(-50%, -50%);
  -webkit-transform: translate(-50%, -50%);
  -o-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
}

.visualization-loading-chart {
  margin: 0;
  position: absolute;
  @extend %center-div;
}

.visualization-error-div {
  overflow: auto;
  position: relative;
  @extend %center-div;
}

.panel-full-width {
  width: 100%;
  height: 100%;
}

.panels-title-text {
  @include euiTextTruncate;
}
