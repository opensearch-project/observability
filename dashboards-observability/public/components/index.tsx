/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import ReactDOM from 'react-dom';
import { QueryManager } from 'common/query_manager';
import { AppMountParameters, CoreStart } from '../../../../src/core/public';
import { AppPluginStartDependencies } from '../types';
import { App } from './app';

export const Observability = (
  CoreStartProp: CoreStart,
  DepsStart: AppPluginStartDependencies,
  AppMountParametersProp: AppMountParameters,
  pplService: any,
  dslService: any,
  savedObjects: any,
  timestampUtils: any,
  queryManager: QueryManager
) => {
  ReactDOM.render(
    <App
      CoreStartProp={CoreStartProp}
      DepsStart={DepsStart}
      pplService={pplService}
      dslService={dslService}
      savedObjects={savedObjects}
      timestampUtils={timestampUtils}
      queryManager={queryManager}
    />,
    AppMountParametersProp.element
  );

  return () => ReactDOM.unmountComponentAtNode(AppMountParametersProp.element);
};
