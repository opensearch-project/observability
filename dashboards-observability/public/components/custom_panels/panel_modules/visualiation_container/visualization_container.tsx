/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import {
  EuiButtonIcon,
  EuiContextMenuItem,
  EuiContextMenuPanel,
  EuiFlexGroup,
  EuiFlexItem,
  EuiIcon,
  EuiLoadingChart,
  EuiPanel,
  EuiPopover,
  EuiSpacer,
  EuiText,
} from '@elastic/eui';
import React, { useEffect, useMemo, useState } from 'react';
import PPLService from '../../../../services/requests/ppl';
import { displayVisualization, getQueryResponse } from '../../helpers/utils';
import './visualization_container.scss';

/*
 * Visualization container - This module is a placeholder to add visualizations in react-grid-layout
 *
 * Props taken in as params are:
 * editMode: boolean to check if the panel is in edit mode
 * visualizationId: unique visualization id
 * visualizationTitle: visualization name
 * query: ppl query to load the visualization
 * pplService: ppl requestor service
 * type: type of visualization [bar, horizontal_bar, line]
 * fromTime: start time in date filter
 * toTime: end time in date filter
 * onRefresh: boolean value to trigger refresh of visualizations
 * cloneVisualization: function to clone a visualization in panel
 * pplFilterValue: string with panel PPL filter value
 * showFlyout: function to show the flyout
 * removeVisualization: function to remove all the visualizations
 */

type Props = {
  editMode: boolean;
  visualizationId: string;
  visualizationTitle: string;
  query: string;
  pplService: PPLService;
  type: string;
  timeField: string;
  fromTime: string;
  toTime: string;
  onRefresh: boolean;
  cloneVisualization: (
    newVisualizationTitle: string,
    pplQuery: string,
    newVisualizationType: string,
    newVisualizationTimeField: string
  ) => void;
  pplFilterValue: string;
  showFlyout: (isReplacement?: boolean | undefined, replaceVizId?: string | undefined) => void;
  removeVisualization: (visualizationId: string) => void;
};

export const VisualizationContainer = ({
  editMode,
  visualizationId,
  visualizationTitle,
  pplService,
  query,
  type,
  timeField,
  fromTime,
  toTime,
  onRefresh,
  cloneVisualization,
  pplFilterValue,
  showFlyout,
  removeVisualization,
}: Props) => {
  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
  const [disablePopover, setDisablePopover] = useState(false);
  const [visualizationData, setVisualizationData] = useState<Plotly.Data[]>([]);
  const [isLoading, setIsLoading] = useState(true);
  const [isError, setIsError] = useState('');
  const onActionsMenuClick = () => setIsPopoverOpen((isPopoverOpen) => !isPopoverOpen);
  const closeActionsMenu = () => setIsPopoverOpen(false);

  const popoverPanel = [
    <EuiContextMenuItem
      key="Edit"
      disabled={disablePopover}
      onClick={() => {
        closeActionsMenu();
        showFlyout(true, visualizationId);
      }}
    >
      Replace
    </EuiContextMenuItem>,
    <EuiContextMenuItem
      key="Duplicate"
      disabled={disablePopover}
      onClick={() => {
        closeActionsMenu();
        cloneVisualization(visualizationTitle, query, type, timeField);
      }}
    >
      Duplicate
    </EuiContextMenuItem>,
  ];

  const loadVisaulization = async () => {
    await getQueryResponse(
      pplService,
      query,
      type,
      fromTime,
      toTime,
      setVisualizationData,
      setIsLoading,
      setIsError,
      pplFilterValue,
      timeField
    );
  };

  const memoisedVisualizationBox = useMemo(
    () => (
      <div className="visualization-div">
        {isLoading ? (
          <EuiLoadingChart size="xl" mono className="visualization-loading-chart" />
        ) : isError != '' ? (
          <div className="visualization-error-div">
            <EuiSpacer size="l" />
            <EuiIcon type="alert" color="danger" size="l" />
            <EuiSpacer size="l" />
            <EuiText>
              <h2>Error in rendering the visualizaiton</h2>
            </EuiText>
            <EuiSpacer size="l" />
            <EuiText>
              <p>{isError}</p>
            </EuiText>
          </div>
        ) : (
          displayVisualization(visualizationData, type)
        )}
      </div>
    ),
    [onRefresh, isLoading, isError, visualizationData, type]
  );

  useEffect(() => {
    loadVisaulization();
  }, [onRefresh]);

  useEffect(() => {
    editMode ? setDisablePopover(true) : setDisablePopover(false);
  }, [editMode]);

  return (
    <EuiPanel className="panel-full-width" grow={false}>
      <div className={editMode ? 'mouseGrabber' : ''}>
        <EuiFlexGroup justifyContent="spaceBetween">
          <EuiFlexItem grow={false}>
            <EuiText grow={false}>
              <h5>{visualizationTitle}</h5>
            </EuiText>
          </EuiFlexItem>
          <EuiFlexItem grow={false}>
            {disablePopover ? (
              <EuiIcon
                type="crossInACircleFilled"
                onClick={() => {
                  removeVisualization(visualizationId);
                }}
              />
            ) : (
              <EuiPopover
                button={
                  <EuiButtonIcon
                    aria-label="actionMenuButton"
                    iconType="boxesHorizontal"
                    onClick={onActionsMenuClick}
                  />
                }
                isOpen={isPopoverOpen}
                closePopover={closeActionsMenu}
                anchorPosition="downLeft"
              >
                <EuiContextMenuPanel items={popoverPanel} />
              </EuiPopover>
            )}
          </EuiFlexItem>
        </EuiFlexGroup>
      </div>
      {memoisedVisualizationBox}
    </EuiPanel>
  );
};
