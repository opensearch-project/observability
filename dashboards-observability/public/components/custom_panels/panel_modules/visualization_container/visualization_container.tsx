/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import {
  EuiButton,
  EuiButtonIcon,
  EuiCodeBlock,
  EuiContextMenuItem,
  EuiContextMenuPanel,
  EuiFlexGroup,
  EuiFlexItem,
  EuiIcon,
  EuiLoadingChart,
  EuiModal,
  EuiModalBody,
  EuiModalFooter,
  EuiModalHeader,
  EuiModalHeaderTitle,
  EuiPanel,
  EuiPopover,
  EuiSpacer,
  EuiText,
  EuiToolTip,
} from '@elastic/eui';
import React, { useEffect, useMemo, useState } from 'react';
import { CoreStart } from '../../../../../../../src/core/public';
import PPLService from '../../../../services/requests/ppl';
import {
  displayVisualization,
  renderCatalogVisualization,
  renderSavedVisualization,
} from '../../helpers/utils';
import './visualization_container.scss';

/*
 * Visualization container - This module is a placeholder to add visualizations in react-grid-layout
 *
 * Props taken in as params are:
 * editMode: boolean to check if the panel is in edit mode
 * visualizationId: unique visualization id
 * visualizationTitle: visualization name
 * query: ppl query to load the visualization
 * pplService: ppl requestor service
 * type: type of visualization [bar, horizontal_bar, line]
 * fromTime: start time in date filter
 * toTime: end time in date filter
 * onRefresh: boolean value to trigger refresh of visualizations
 * cloneVisualization: function to clone a visualization in panel
 * pplFilterValue: string with panel PPL filter value
 * showFlyout: function to show the flyout
 * removeVisualization: function to remove all the visualizations
 * catalogVisualization: boolean pointing if the container is used for catalog metrics
 * spanParam: Override the span(timestamp, 1h) in visualization to span(timestamp, spanParam)
 */

interface Props {
  http: CoreStart['http'];
  editMode: boolean;
  visualizationId: string;
  savedVisualizationId: string;
  pplService: PPLService;
  fromTime: string;
  toTime: string;
  onRefresh: boolean;
  pplFilterValue: string;
  usedInNotebooks?: boolean;
  onEditClick: (savedVisualizationId: string) => any;
  cloneVisualization?: (visualzationTitle: string, savedVisualizationId: string) => void;
  showFlyout?: (isReplacement?: boolean | undefined, replaceVizId?: string | undefined) => void;
  removeVisualization?: (visualizationId: string) => void;
  catalogVisualization?: boolean;
  spanParam?: string;
}

export const VisualizationContainer = ({
  http,
  editMode,
  visualizationId,
  savedVisualizationId,
  pplService,
  fromTime,
  toTime,
  onRefresh,
  pplFilterValue,
  usedInNotebooks,
  onEditClick,
  cloneVisualization,
  showFlyout,
  removeVisualization,
  catalogVisualization,
  spanParam,
}: Props) => {
  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
  const [disablePopover, setDisablePopover] = useState(false);
  const [visualizationTitle, setVisualizationTitle] = useState('');
  const [visualizationType, setVisualizationType] = useState('');
  const [visualizationMetaData, setVisualizationMetaData] = useState();
  const [visualizationData, setVisualizationData] = useState<Plotly.Data[]>([]);
  const [isLoading, setIsLoading] = useState(true);
  const [isError, setIsError] = useState('');
  const onActionsMenuClick = () => setIsPopoverOpen((currPopoverOpen) => !currPopoverOpen);
  const closeActionsMenu = () => setIsPopoverOpen(false);

  const [isModalVisible, setIsModalVisible] = useState(false);

  const closeModal = () => setIsModalVisible(false);
  const showModal = () => setIsModalVisible(true);

  let modal;

  if (isModalVisible) {
    modal = (
      <EuiModal onClose={closeModal}>
        <EuiModalHeader>
          <EuiModalHeaderTitle>
            <h1>{visualizationMetaData.name}</h1>
          </EuiModalHeaderTitle>
        </EuiModalHeader>

        <EuiModalBody>
          This PPL Query is generated in runtime from selected data source
          <EuiSpacer />
          <EuiCodeBlock language="html" isCopyable>
            {visualizationMetaData.query}
          </EuiCodeBlock>
        </EuiModalBody>

        <EuiModalFooter>
          <EuiButton onClick={closeModal} fill>
            Close
          </EuiButton>
        </EuiModalFooter>
      </EuiModal>
    );
  }

  let popoverPanel = [
    <EuiContextMenuItem
      data-test-subj="editVizContextMenuItem"
      key="Edit"
      disabled={disablePopover}
      onClick={() => {
        closeActionsMenu();
        onEditClick(savedVisualizationId);
      }}
    >
      Edit
    </EuiContextMenuItem>,
    <EuiContextMenuItem
      key="Replace"
      disabled={disablePopover}
      onClick={() => {
        closeActionsMenu();
        showFlyout(true, visualizationId);
      }}
    >
      Replace
    </EuiContextMenuItem>,
    <EuiContextMenuItem
      key="Duplicate"
      disabled={disablePopover}
      onClick={() => {
        closeActionsMenu();
        cloneVisualization(visualizationTitle, savedVisualizationId);
      }}
    >
      Duplicate
    </EuiContextMenuItem>,
  ];

  let showModelPanel = [
    <EuiContextMenuItem
      data-test-subj="showCatalogPPLQuery"
      key="view_query"
      disabled={disablePopover}
      onClick={() => {
        closeActionsMenu();
        showModal();
      }}
    >
      View query
    </EuiContextMenuItem>,
  ];

  if (usedInNotebooks) {
    popoverPanel = catalogVisualization ? [showModelPanel] : [popoverPanel[0]];
  }

  const loadVisaulization = async () => {
    if (catalogVisualization)
      await renderCatalogVisualization(
        http,
        pplService,
        savedVisualizationId,
        fromTime,
        toTime,
        pplFilterValue,
        spanParam,
        setVisualizationTitle,
        setVisualizationType,
        setVisualizationData,
        setVisualizationMetaData,
        setIsLoading,
        setIsError
      );
    else
      await renderSavedVisualization(
        http,
        pplService,
        savedVisualizationId,
        fromTime,
        toTime,
        pplFilterValue,
        spanParam,
        setVisualizationTitle,
        setVisualizationType,
        setVisualizationData,
        setVisualizationMetaData,
        setIsLoading,
        setIsError
      );
  };

  const memoisedVisualizationBox = useMemo(
    () => (
      <div className="visualization-div">
        {isLoading ? (
          <EuiLoadingChart size="xl" mono className="visualization-loading-chart" />
        ) : isError !== '' ? (
          <div className="visualization-error-div">
            <EuiIcon type="alert" color="danger" size="s" />
            <EuiSpacer size="s" />
            <EuiText size="s">
              <p>{isError}</p>
            </EuiText>
          </div>
        ) : (
          displayVisualization(visualizationMetaData, visualizationData, visualizationType)
        )}
      </div>
    ),
    [onRefresh, isLoading, isError, visualizationData, visualizationType, visualizationMetaData]
  );

  useEffect(() => {
    loadVisaulization();
  }, [onRefresh]);

  useEffect(() => {
    editMode ? setDisablePopover(true) : setDisablePopover(false);
  }, [editMode]);

  return (
    <>
      <EuiPanel
        data-test-subj={`${visualizationTitle}VisualizationPanel`}
        className="panel-full-width"
        grow={false}
      >
        <div className={editMode ? 'mouseGrabber' : ''}>
          <EuiFlexGroup justifyContent="spaceBetween">
            <EuiFlexItem
              style={{
                width: '35%',
              }}
            >
              <EuiText grow={false} className="panels-title-text">
                <EuiToolTip delay="long" position="top" content={visualizationTitle}>
                  <h5>{visualizationTitle}</h5>
                </EuiToolTip>
              </EuiText>
            </EuiFlexItem>
            <EuiFlexItem grow={false} className="visualization-action-button">
              {disablePopover ? (
                <EuiIcon
                  type="crossInACircleFilled"
                  onClick={() => {
                    removeVisualization(visualizationId);
                  }}
                />
              ) : (
                <EuiPopover
                  button={
                    <EuiButtonIcon
                      aria-label="actionMenuButton"
                      iconType="boxesHorizontal"
                      onClick={onActionsMenuClick}
                    />
                  }
                  isOpen={isPopoverOpen}
                  closePopover={closeActionsMenu}
                  anchorPosition="downLeft"
                >
                  <EuiContextMenuPanel items={popoverPanel} />
                </EuiPopover>
              )}
            </EuiFlexItem>
          </EuiFlexGroup>
        </div>
        {memoisedVisualizationBox}
      </EuiPanel>
      {modal}
    </>
  );
};
