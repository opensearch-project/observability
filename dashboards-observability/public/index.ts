/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import './components/trace_analytics/index.scss';
import './components/notebooks/index.scss'

import { PluginInitializer, PluginInitializerContext } from '../../../src/core/public';
import {
  ObservabilityPlugin
} from './plugin';

export { ObservabilityPlugin as Plugin };

export const plugin = (initializerContext: PluginInitializerContext) => new ObservabilityPlugin(initializerContext);