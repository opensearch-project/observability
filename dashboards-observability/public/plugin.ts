/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { AppMountParameters, CoreSetup, CoreStart, Plugin } from '../../../src/core/public';
import {
  observabilityID,
  observabilityPluginOrder,
  observabilityTitle,
} from '../common/constants/shared';
import PPLService from './services/requests/ppl';
import DSLService from './services/requests/dsl';
import TimestampUtils from './services/timestamp/timestamp';
import SavedObjects from './services/saved_objects/event_analytics/saved_objects';
import { AppPluginStartDependencies, ObservabilitySetup, ObservabilityStart } from './types';
import { convertLegacyNotebooksUrl } from './components/notebooks/components/helpers/legacy_route_helpers';
import { convertLegacyTraceAnalyticsUrl } from './components/trace_analytics/components/common/legacy_route_helpers';
import { uiSettingsService } from '../common/utils';
import { QueryManager } from '../common/query_manager';
export class ObservabilityPlugin implements Plugin<ObservabilitySetup, ObservabilityStart> {
  public setup(core: CoreSetup): ObservabilitySetup {
    uiSettingsService.init(core.uiSettings, core.notifications);

    // redirect legacy notebooks URL to current URL under observability
    if (window.location.pathname.includes('notebooks-dashboards')) {
      window.location.assign(convertLegacyNotebooksUrl(window.location));
    }

    // redirect legacy trace analytics URL to current URL under observability
    if (window.location.pathname.includes('trace-analytics-dashboards')) {
      window.location.assign(convertLegacyTraceAnalyticsUrl(window.location));
    }

    core.application.register({
      id: observabilityID,
      title: observabilityTitle,
      category: {
        id: 'opensearch',
        label: 'OpenSearch Plugins',
        order: 2000,
      },
      order: observabilityPluginOrder,
      async mount(params: AppMountParameters) {
        const { Observability } = await import('./components/index');
        const [coreStart, depsStart] = await core.getStartServices();
        const pplService = new PPLService(coreStart.http);
        const dslService = new DSLService(coreStart.http);
        const savedObjects = new SavedObjects(coreStart.http);
        const timestampUtils = new TimestampUtils(dslService, pplService);
        const qm = new QueryManager();
        return Observability(
          coreStart,
          depsStart as AppPluginStartDependencies,
          params,
          pplService,
          dslService,
          savedObjects,
          timestampUtils,
          qm
        );
      },
    });

    // Return methods that should be available to other plugins
    return {};
  }
  public start(core: CoreStart): ObservabilityStart {
    return {};
  }
  public stop() {}
}
