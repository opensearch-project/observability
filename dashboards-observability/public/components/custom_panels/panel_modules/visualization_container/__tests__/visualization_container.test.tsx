/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import PPLService from '../../../../../services/requests/ppl';
import React from 'react';
import { VisualizationContainer } from '../visualization_container';
import httpClientMock from '../../../../../../test/__mocks__/httpClientMock';
import { HttpResponse } from '../../../../../../../../src/core/public';
import { waitFor } from '@testing-library/react';
import {
  sampleSavedVisualization,
  samplePPLResponse,
} from '../../../../../../test/panels_constants';

describe('Visualization Container Component', () => {
  configure({ adapter: new Adapter() });

  it('renders add visualization container', async () => {
    httpClientMock.get = jest.fn(() =>
      Promise.resolve((sampleSavedVisualization as unknown) as HttpResponse)
    );

    httpClientMock.post = jest.fn(() =>
      Promise.resolve((samplePPLResponse as unknown) as HttpResponse)
    );

    const editMode = true;
    const visualizationId = 'panel_viz_9b1deb4d-3b7d-4bad-9bdd-2b0d7b3dcb6d';
    const savedVisualizationId = 'oiuccXwBYVazWqOO1e06';
    const fromTime = 'now-15m';
    const toTime = 'now';
    const onRefresh = true;
    const cloneVisualization = jest.fn();
    const pplFilterValue = 'where Carrier = "OpenSearch-Air"';
    const showFlyout = jest.fn();
    const removeVisualization = jest.fn();
    const pplService = new PPLService(httpClientMock);

    const wrapper = mount(
      <VisualizationContainer
        http={httpClientMock}
        editMode={editMode}
        visualizationId={visualizationId}
        savedVisualizationId={savedVisualizationId}
        pplService={pplService}
        fromTime={fromTime}
        toTime={toTime}
        onRefresh={onRefresh}
        cloneVisualization={cloneVisualization}
        pplFilterValue={pplFilterValue}
        showFlyout={showFlyout}
        removeVisualization={removeVisualization}
      />
    );
    wrapper.update();

    await waitFor(() => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
