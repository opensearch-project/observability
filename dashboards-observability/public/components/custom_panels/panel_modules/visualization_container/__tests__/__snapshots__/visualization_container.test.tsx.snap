// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Visualization Container Component renders add visualization container 1`] = `
<VisualizationContainer
  cloneVisualization={[MockFunction]}
  editMode={true}
  fromTime="now-15m"
  http={[MockFunction]}
  onEditClick={[Function]}
  onRefresh={true}
  pplFilterValue="where Carrier = \\"OpenSearch-Air\\""
  pplService={
    PPLService {
      "fetch": [Function],
      "http": [MockFunction],
    }
  }
  removeVisualization={[MockFunction]}
  savedVisualizationId="oiuccXwBYVazWqOO1e06"
  showFlyout={[MockFunction]}
  toTime="now"
  visualizationId="panel_viz_9b1deb4d-3b7d-4bad-9bdd-2b0d7b3dcb6d"
>
  <EuiPanel
    className="panel-full-width"
    data-test-subj="VisualizationPanel"
    grow={false}
  >
    <div
      className="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiPanel--flexGrowZero panel-full-width"
      data-test-subj="VisualizationPanel"
    >
      <div
        className="mouseGrabber"
      >
        <EuiFlexGroup
          justifyContent="spaceBetween"
        >
          <div
            className="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
          >
            <EuiFlexItem
              style={
                Object {
                  "width": "35%",
                }
              }
            >
              <div
                className="euiFlexItem"
                style={
                  Object {
                    "width": "35%",
                  }
                }
              >
                <EuiText
                  className="panels-title-text"
                  grow={false}
                >
                  <div
                    className="euiText euiText--medium panels-title-text euiText--constrainedWidth"
                  >
                    <EuiToolTip
                      content=""
                      delay="long"
                      position="top"
                    >
                      <span
                        className="euiToolTipAnchor"
                        onKeyUp={[Function]}
                        onMouseOut={[Function]}
                        onMouseOver={[Function]}
                      >
                        <h5
                          onBlur={[Function]}
                          onFocus={[Function]}
                        />
                      </span>
                    </EuiToolTip>
                  </div>
                </EuiText>
              </div>
            </EuiFlexItem>
            <EuiFlexItem
              className="visualization-action-button"
              grow={false}
            >
              <div
                className="euiFlexItem euiFlexItem--flexGrowZero visualization-action-button"
              >
                <EuiIcon
                  onClick={[Function]}
                  type="crossInACircleFilled"
                >
                  <EuiIconEmpty
                    aria-hidden={true}
                    className="euiIcon euiIcon--medium euiIcon-isLoading"
                    focusable="false"
                    onClick={[Function]}
                    role="img"
                    style={null}
                  >
                    <svg
                      aria-hidden={true}
                      className="euiIcon euiIcon--medium euiIcon-isLoading"
                      focusable="false"
                      height={16}
                      onClick={[Function]}
                      role="img"
                      style={null}
                      viewBox="0 0 16 16"
                      width={16}
                      xmlns="http://www.w3.org/2000/svg"
                    />
                  </EuiIconEmpty>
                </EuiIcon>
              </div>
            </EuiFlexItem>
          </div>
        </EuiFlexGroup>
      </div>
      <div
        className="visualization-div"
      >
        <EuiLoadingChart
          className="visualization-loading-chart"
          mono={true}
          size="xl"
        >
          <span
            className="euiLoadingChart euiLoadingChart--mono visualization-loading-chart euiLoadingChart--xLarge"
          >
            <span
              className="euiLoadingChart__bar"
            />
            <span
              className="euiLoadingChart__bar"
            />
            <span
              className="euiLoadingChart__bar"
            />
            <span
              className="euiLoadingChart__bar"
            />
          </span>
        </EuiLoadingChart>
      </div>
    </div>
  </EuiPanel>
</VisualizationContainer>
`;
